server/mobile-services-info.js

```diff
@@ -37,7 +48,92 @@ const PushService = {
+    const getAndroidVariant = kubeclient.apis[androidVariantCRD.spec.group].v1alpha1
+      .namespace(namespace)
+      .androidvariants(`${appname}${ANDROID_UPS_SUFFIX}`)
+      .get()
+      .then(resp => resp.body)
+      .catch(err => {
+        const name = `${appname}${ANDROID_UPS_SUFFIX}`;
+        if (err && err.statusCode && err.statusCode === 404) {
+          console.info(`Can not find AndroidVariant ${name}`);
+        } else {
+          console.warn(`Error when fetch AndroidVariant ${name}`, err);
+        }
+      });
+
+    // get the iOS variant associated with this app
+    const getIOSVariant = kubeclient.apis[iosVariantCRD.spec.group].v1alpha1
+      .namespace(namespace)
+      .iosvariants(`${appname}${IOS_UPS_SUFFIX}`)
+      .get()
+      .then(resp => resp.body)
+      .catch(err => {
+        const name = `${appname}${IOS_UPS_SUFFIX}`;
+        if (err && err.statusCode && err.statusCode === 404) {
+          console.info(`Can not find IOSVariant ${name}`);
+        } else {
+          console.warn(`Error when fetch IOSVariant ${name}`, err);
+        }
+      });
+
+    return kubeclient.apis[pushApplicationCRD.spec.group].v1alpha1
+      .namespace(namespace)
+      .pushapplications(appname)
+      .get()
+      .then(resp => resp.body)
+      .then(pushApplication => ({
+        id: pushApplication.metadata.uid,
+        pushApplicationId: pushApplication.status.pushApplicationId,
+        name: PUSH_SERVICE_TYPE,
+        type: PUSH_SERVICE_TYPE,
+        variants: []
+      }))
+      .then(config =>
+        Promise.all([getAndroidVariant, getIOSVariant])
+          .then(variants => variants.filter(Boolean))
+          .then(variants => {
+            if (!variants || !variants.length) {
+              return null;
+            }
+
+            // get the AndroidVariant and add it
+            const androidVariant = variants.find(v => !!v && v.kind === ANDROID_VARIANT_KIND);
+            if (androidVariant && androidVariant.status) {
+              config.variants.push({
+                id: androidVariant.metadata.uid,
+                name: ANDROID_VARIANT_TYPE,
+                type: ANDROID_VARIANT_TYPE,
+                secret: androidVariant.status.secret,
+                variantId: androidVariant.status.variantId,
+                pushApplicationId: androidVariant.spec.pushApplicationId
+              });
+            }
+
+            // get the IOSVariant and add it
+            const iosVariant = variants.find(v => v.kind === IOS_VARIANT_KIND);
+            if (iosVariant && iosVariant.status) {
+              config.variants.push({
+                id: iosVariant.metadata.uid,
+                name: IOS_VARIANT_TYPE,
+                type: IOS_VARIANT_TYPE,
+                secret: iosVariant.status.secret,
+                variantId: iosVariant.status.variantId,
+                pushApplicationId: iosVariant.spec.pushApplicationId
+              });
+            }
+
+            return config;
```

Comment:
the expected format of the config by the client SDK should be like this:

```
{
  id: 'any id',
  name: 'push',
  type: 'push',
  url: <push server url>,
  config: {
    'android': {
        'variantId': <id of the variant>,
        'variantSecret': <secret>
     },
    'ios': {
        'variantId': <id of the variant>,
        'variantSecret': <secret>
     },
  }
}
```

I think the current returned config doesn't match this format, can we update it?

Reply:
Will do! :+1:

Reply:
Updated now:

![Screenshot from 2019-07-17 14-57-56](https://user-images.githubusercontent.com/11743717/61381838-8ff75f00-a8a3-11e9-89da-d0433de2464e.png)